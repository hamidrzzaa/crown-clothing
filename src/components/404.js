import React from "react";
import { Link } from "react-router-dom";

export const PageNotFound = () => (
  <div>
    <h1>
      404! -<Link to="/">Go Home</Link>
    </h1>
  </div>
);

export default PageNotFound;
