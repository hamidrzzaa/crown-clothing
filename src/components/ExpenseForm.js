import React from "react";
import moment from "moment";
import "react-dates/initialize";
import { SingleDatePicker } from "react-dates";
import "react-dates/lib/css/_datepicker.css";
import { connect } from "react-redux";

import { addExpense } from "../actions/expenses";

export class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      note: props.expense ? props.expense.note : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      focused: false,
      error: ""
    };
  }
  onInputChange = e => {
    // const inputName = e.target.name;
    // const inputValue = e.target.value;
    e.persist();
    if (e.target.name === "amount") {
      const amount = e.target.value;
      if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
        this.setState(() => ({ amount: amount }));
      }
    } else {
      this.setState(() => ({ [e.target.name]: e.target.value }));
    }
  };

  onDateChange = date => {
    if (date) {
      this.setState(() => ({ createdAt: date }));
    }
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ focused: focused }));
  };

  onFormSubmit = e => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      return this.setState(() => ({
        error: "please provide the description or amount"
      }));
    }
    this.props.onSubmit({
      description: this.state.description,
      amount: parseFloat(this.state.amount, 10),
      createdAt: this.state.createdAt.valueOf(),
      note: this.state.note
    });
    this.setState(() => ({ error: "", description: "", amount: "", note: "" }));
  };
  renderError() {
    if (this.state.error) {
      return (
        <div style={{ color: "red" }}>
          <p>{this.state.error}</p>
        </div>
      );
    } else {
      return undefined;
    }
  }
  render() {
    return (
      <div>
        {this.renderError()}
        <form onSubmit={this.onFormSubmit}>
          <input
            onChange={this.onInputChange}
            name="description"
            type="text"
            placeholder="simple description..."
            autoFocus
            value={this.state.description}
          />
          <input
            onChange={this.onInputChange}
            value={this.state.amount}
            name="amount"
            type="text"
            placeholder="Enter the Amount "
          />
          <SingleDatePicker
            id="wolf is here"
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            onFocusChange={this.onFocusChange}
            focused={this.state.focused}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            onChange={this.onInputChange}
            value={this.state.note}
            name="note"
            placeholder="Note about the Expense"
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}

ExpenseForm.displayName = "ExpenseForm";

export default connect(null, { addExpense })(ExpenseForm);
