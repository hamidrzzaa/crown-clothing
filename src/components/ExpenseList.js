import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";

export const ExpenseList = props => {
  return (
    <div>
      {props.expenses.length === 0 ? (
        <p>No Expenses</p>
      ) : (
        props.expenses.map(expense => (
          <ExpenseListItem
            key={expense.id}
            description={expense.description}
            amount={expense.amount}
            createdAt={expense.createdAt}
            id={expense.id}
          />
        ))
      )}
    </div>
  );
};

const mapStateToProps = state => {
  return { expenses: selectExpenses(state.expenses, state.filters) };
};

export default connect(mapStateToProps)(ExpenseList);
