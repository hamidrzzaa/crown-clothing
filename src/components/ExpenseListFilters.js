import React from "react";
import { connect } from "react-redux";
import {
  setFilterText,
  sortByAmount,
  sortByDate,
  setEndDate,
  setStartDate
} from "../actions/filters";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilter extends React.Component {
  state = { calenderFocused: null };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = focusedInput => {
    this.setState(() => ({ calenderFocused: focusedInput }));
  };
  onTextChange = e => {
    this.props.setFilterText(e.target.value);
  };
  onSelectChange = e => {
    if (e.target.value === "amount") {
      this.props.sortByAmount();
    } else {
      this.props.sortByDate();
    }
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSelectChange}
        >
          <option value="amount">Amount</option>
          <option value="date">Date</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps, {
  setFilterText,
  sortByAmount,
  sortByDate,
  setEndDate,
  setStartDate
})(ExpenseListFilter);
