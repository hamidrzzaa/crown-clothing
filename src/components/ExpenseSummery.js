import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
export const ExpenseSummery = ({expenseCount,expensesTotal}) => {
    const expenseWord=expenseCount===1?'expense':'expenses'
    const format=numeral(expensesTotal).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} total is {format}</h1>
        </div>
    )
}

const mapStateToProps=(state)=>{
   const visibleExpenses=selectExpenses(state.expenses,state.filters)
    return {
       expenseCount:visibleExpenses.length,
        expensesTotal:selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummery)