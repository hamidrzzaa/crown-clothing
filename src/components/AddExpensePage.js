import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { addExpense } from "../actions/expenses";

export const AddExpensePage = props => {
  const onSubmit = data => {
    props.addExpense(data);
    props.history.push("/");
  };
  return (
    <div>
      <h1>Add Expense</h1>
      <ExpenseForm onSubmit={onSubmit} />
    </div>
  );
};

// const mapDispatchToProps = dispatch => ({
//   addExpense: data => dispatch(addExpense(data))
// });

export default connect(null, { addExpense })(AddExpensePage);
