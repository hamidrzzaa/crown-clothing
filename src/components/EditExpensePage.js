import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";
export const EditExpensePage = props => {
  const onFromSubmit = date => {
    props.editExpense(props.expense.id, date);
    props.history.push("/");
  };

  const onExpenseDelete = () => {
    props.removeExpense(props.expense.id);
    props.history.push("/");
  };

  if (props.expense) {
    return (
      <div>
        <ExpenseForm onSubmit={onFromSubmit} expense={props.expense} />
        <button onClick={onExpenseDelete}>Remove</button>
      </div>
    );
  } else {
    return (
      <div>
        <p>There is no expense </p>
      </div>
    );
  }
};

const mapStateToProps = (state, ownProps) => {
  const idParam = ownProps.match.params.id;
  const filter = state.expenses.filter(exp => exp.id === idParam);

  return {
    expense: filter[0]
  };
};

export default connect(mapStateToProps, { editExpense, removeExpense })(
  EditExpensePage
);
