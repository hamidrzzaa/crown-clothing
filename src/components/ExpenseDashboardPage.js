import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilter from "./ExpenseListFilters";
import ExpenseSummery from "./ExpenseSummery";

export const ExpenseDashboardPage = () => {
    return (
        <div>
            <ExpenseListFilter/>
            <ExpenseSummery/>
            <ExpenseList/>
        </div>
    );
};

export default ExpenseDashboardPage;
