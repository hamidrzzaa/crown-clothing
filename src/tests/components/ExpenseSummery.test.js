import React from "react";
import {shallow} from "enzyme";
import {ExpenseSummery} from "../../components/ExpenseSummery";
import toJson from "enzyme-to-json";

test("should correctly render expense summery", () => {
    const wrapper = shallow(
        <ExpenseSummery expenseCount={1} expenseTotal={12.0}/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
});
test("should correctly render expense summery with multiple expenses", () => {
    const wrapper = shallow(
        <ExpenseSummery expenseCount={4} expenseTotal={12.0}/>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
});
