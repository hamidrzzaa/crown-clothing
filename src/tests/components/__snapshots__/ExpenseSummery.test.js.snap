// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expense summery 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     total is 
    $0.00
  </h1>
</div>
`;

exports[`should correctly render expense summery with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    4
     
    expenses
     total is 
    $0.00
  </h1>
</div>
`;
